ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/Tree.java

```diff
@@ -236,7 +223,7 @@ private boolean improvedEnough(Leaf leaf) {
     private RaoResult buildOutput() {
         RaoResult raoResult = new RaoResult(optimalLeaf.getStatus().equals(Leaf.Status.ERROR) ? RaoResult.Status.FAILURE : RaoResult.Status.SUCCESS);
         raoResult.setPreOptimVariantId(rootLeaf.getInitialVariantId());
-        raoResult.setPostOptimVariantId(optimalLeaf.getBestVariantId());
+        raoResult.getPostOptimVariantIdPerStateId().put(optimalLeaf.getRaoData().getOptimizedState().getId(), optimalLeaf.getBestVariantId());
```

Comment:
same as another comment, seems better to do the filling of the map in the class that has direct access the map

Reply:
done